From 5d24ba00a7735ab3b7b381a102dc5536c12c8133 Mon Sep 17 00:00:00 2001
From: Ernestas Kulik <ernestask@gnome.org>
Date: Sun, 7 Oct 2018 11:40:41 +0300
Subject: [PATCH 1/2] build: Fix libcue version comparison
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

It currently performs a string comparison, as opposed to using
version_compare(), which leads to results like
“unknown >= 2.0.0 == True” when the dependency was not found.

Fixes https://gitlab.gnome.org/GNOME/tracker-miners/issues/32

(cherry picked from commit 6e281037a5f0abc12e5e8902e445f264200980b0)
---
 meson.build | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/meson.build b/meson.build
index 4273638dd..c30ada4ed 100644
--- a/meson.build
+++ b/meson.build
@@ -317,7 +317,7 @@ conf.set('GSTREAMER_BACKEND_DISCOVERER', gstreamer_backend_name == 'Discoverer')
 conf.set('GSTREAMER_BACKEND_GUPNP_DLNA', gstreamer_backend_name == 'GUPnP-DLNA')
 conf.set('HAVE_HAL', battery_detection_library_name == 'hal')
 conf.set('HAVE_LIBCUE', libcue.found())
-conf.set('HAVE_LIBCUE2', libcue.version() >= '2.0.0')
+conf.set('HAVE_LIBCUE2', libcue.version().version_compare('>= 2.0.0'))
 conf.set('HAVE_LIBICU_CHARSET_DETECTION', charset_library_name == 'icu')
 conf.set('HAVE_LIBEXIF', libexif.found())
 conf.set('HAVE_LIBIPTCDATA', libiptcdata.found())
-- 
2.17.0


From ff6a076330ba1b5a383196795b18e2c069ac54c6 Mon Sep 17 00:00:00 2001
From: Sam Thursfield <sam@afuera.me.uk>
Date: Sun, 7 Oct 2018 23:42:28 +0200
Subject: [PATCH 2/2] build: Fix breakage when libcue isn't found

Commit 6e281037a5f0abc introduced a regression on systems where libcue
is not available.

Fixes https://gitlab.gnome.org/GNOME/tracker-miners/issues/33

(cherry picked from commit fd0eb3cd98f279591cbe8635eadf42027dd5ddae)
---
 meson.build | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/meson.build b/meson.build
index c30ada4ed..28072646a 100644
--- a/meson.build
+++ b/meson.build
@@ -317,7 +317,7 @@ conf.set('GSTREAMER_BACKEND_DISCOVERER', gstreamer_backend_name == 'Discoverer')
 conf.set('GSTREAMER_BACKEND_GUPNP_DLNA', gstreamer_backend_name == 'GUPnP-DLNA')
 conf.set('HAVE_HAL', battery_detection_library_name == 'hal')
 conf.set('HAVE_LIBCUE', libcue.found())
-conf.set('HAVE_LIBCUE2', libcue.version().version_compare('>= 2.0.0'))
+conf.set('HAVE_LIBCUE2', libcue.found() and libcue.version().version_compare('>= 2.0.0'))
 conf.set('HAVE_LIBICU_CHARSET_DETECTION', charset_library_name == 'icu')
 conf.set('HAVE_LIBEXIF', libexif.found())
 conf.set('HAVE_LIBIPTCDATA', libiptcdata.found())
-- 
2.17.0

